public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -28,9 +29,66 @@ import {
 } from '../../../store';
 import { enrichPresetWorkflowWithUiMetadata } from './utils';
 import { getDataSourceId } from '../../../utils';
+import { getSavedObjectsClient } from '../../../services';
+import semver from 'semver';
 
 interface NewWorkflowProps {}
 
+export const getEffectiveVersion = async (
```

Comment:
suggest to add a comment on why the default is 2.17.0 above this fn signature. Essentially, 2.17 is the minimum version supported.

Speaking of which, we should update the [compatibility](https://github.com/opensearch-project/dashboards-flow-framework/blob/main/opensearch_dashboards.json#L17) to have a minimum version of 2.17, instead of 2.18.

And lastly, for 2.18, we should display the 2.17-specific processors. This leaves us with essentially 3 different states to handle based on the datasource version:
1. <2.17: plugin is not available, don't need to do anything
2. >=2.17, <2.19: plugin is available, only show the neural-search-related usecases & processors
3. >=2.19: plugin is available, show all use cases/processors except the neural-search-related ones.